modules/container-service/managed-cluster/main.bicep

```diff
@@ -596,6 +602,12 @@ resource managedCluster 'Microsoft.ContainerService/managedClusters@2023-07-02-p
       userAssignedIdentityExceptions: podIdentityProfileUserAssignedIdentityExceptions
     }
     securityProfile: {
+      azureKeyVaultKms: enableAzureKeyVaultKms ? {
+        enabled: enableAzureKeyVaultKms
+        keyId: keyVaultKms.keyId
+        keyVaultNetworkAccess: keyVaultKms.keyVaultNetworkAccess
+        keyVaultResourceId: keyVaultKms.keyVaultNetworkAccess == 'Private' ? keyVaultKms.keyVaultResourceId : null
+      } : null
```

Comment:
Hey @aadev1 this looks very much like AKS's version of customer manged keys. For that we actually have a template to keep the interface somewhat consistent across the modules.

This leaves us with 2 options. Either, we merge your PR in and I updated it right after - or - you give the interface a spin and reach out if you need help ;)

The interface should be similar to the following: https://azure.github.io/Azure-Verified-Modules/specs/shared/interfaces/#customer-managed-keys

Note however, it's just an example. Resource-specific adjustments must be implemented per module. In this case, it would look somewhat along the lines of (note, the example requires the other elements of the interface to work):

```suggestion
      azureKeyVaultKms: !empty(customerManagedKey) ? {
        enabled: true
        keyId: !empty(customerManagedKey.?keyVersion ?? '') ? '${cMKKeyVault::cMKKey.properties.keyUri}/${customerManagedKey!.keyVersion}' : cMKKeyVault::cMKKey.properties.keyUriWithVersion
        keyVaultNetworkAccess: customerManagedKey.keyVaultNetworkAccess
        keyVaultResourceId: customerManagedKey.keyVaultNetworkAccess == 'Private' ? cMKKeyVault.id : null
      } : null
```

Hey @jtracey93 & @eriqua , regarding the naming, would you suggest to go with the API's `azureKeyVaultKms` (and just name the CMK interface in this module like it, like @aadev1 did), or should it follow the naming of all other modules using CMK (e.g. batch, cognitive services, etc.)

Reply:
Why should this be different than what done for other modules?

Reply:
I was just wondering (as the name is not the generic `encryption` if people would struggle to find that `customerManagedKey` refers to `azureKeyVaultKms`. Either should work, but I need optinions :D Feel free to jump in too, @aadev1 . The question is what's more important, consistency with other modules, or the consistency with the individual resource provider. Either works.

Reply:
Hi @AlexanderSehr,  We would like to go with option 1 and try and get this change merged as we are using CARML bicep modules for out setup.

Reply:
Alright then. I created a branch upstream and rebased your PR. Will merge it in an pick the topic up from there. Thanks for the contribution as always :)

Reply:
@AlexanderSehr the question is if other resources are consistent with the API naming or not. If not I'd suggest staying consistent with the schema. If this is the first case we should discuss this and update the specs if applicable. I'd anyway tend to align with the schema anyway.

Reply:
Dang, it appears your fork is out of sync and shows conflicts - as such I cannot merge the PR in. Any chance you could update your fork from upstream and resolve those conflicts? The PR can stay open as is.

The only alternative I see is to copy your changes into the target branch, but then the history would be lost.

Reply:
@AlexanderSehr I will speak to one of my colleagues tomorrow about syncing the fork and will get back to you.

Reply:
Hi @AlexanderSehr, I've updated our fork and resolved the merge conflicts and re-ran our tests to make sure it's all working as expected.

Reply:
Awesome, thanks :) Will merge and continue working on it.